specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json

```diff
@@ -464,559 +464,2343 @@
+                    "description": "A message describing the error, intended to be suitable for display in a user interface."
+                }
+            }
+        },
+        "AlertRulesList": {
+            "description": "List all the alert rules.",
+            "properties": {
+                "nextLink": {
+                    "readOnly": true,
+                    "description": "URL to fetch the next set of alert rules.",
+                    "type": "string"
+                },
+                "value": {
+                    "description": "Array of alert rules.",
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/AlertRule"
+                    }
+                }
+            },
+            "required": [
+                "value"
+            ]
+        },
+        "AlertRule": {
+            "type": "object",
+            "description": "Alert rule.",
+            "discriminator": "kind",
+            "allOf": [
+                {
+                    "$ref": "#/definitions/Resource"
+                },
+                {
+                    "$ref": "#/definitions/AlertRuleKind"
+                }
+            ],
+            "properties": {
+                "etag": {
+                    "type": "string",
+                    "description": "Etag of the alert rule."
+                }
+            }
+        },
+        "AlertRuleKind": {
+            "type": "object",
+            "description": "Describes an Azure resource with kind.",
+            "properties": {
+                "kind": {
+                    "type": "string",
+                    "description": "The kind of the alert rule",
+                    "enum": [
+                        "Scheduled"
+                    ],
+                    "x-ms-enum": {
+                        "name": "AlertRuleKind",
+                        "modelAsString": true,
+                        "values": [
+                            {
+                                "value": "Scheduled"
+                            }
+                        ]
+                    }
+                }
+            }
+        },
+        "ScheduledAlertRule": {
+            "x-ms-discriminator-value": "Scheduled",
+            "type": "object",
+            "description": "Represents scheduled alert rule.",
+            "allOf": [
+                {
+                    "$ref": "#/definitions/AlertRule"
+                }
+            ],
+            "properties": {
+                "properties": {
+                    "x-ms-client-flatten": true,
+                    "description": "Scheduled alert rule properties",
+                    "$ref": "#/definitions/ScheduledAlertRuleProperties"
+                }
+            }
+        },
+        "ScheduledAlertRuleProperties": {
+            "type": "object",
+            "description": "Alert rule property bag.",
+            "properties": {
+                "ruleName": {
```

Comment:
ruleName comes from the URL and is also a top level property. Why is this needed?

Reply:
The ruleName is a display name that the customer is giving to the alert, it has spaces and have no restrictions.
The top level "name" property is the one that any resource has in the system, we are planing to generate a guid from our UX since it easier to handle as unique identifier of the resource.